### PR TITLE
feat: improve swapper types

### PR DESCRIPTION
**Pull request by kaladinlight**
- swapper specific supported chainId and chainAdapter types
- use base adapters wherever common `CHAIN_NAMESPACE` logic exists
- update explicit cosmos types to cosmossdk types
- update cosmos/bitcoin naming to cosmossdk/utxo and move up one level to reduce the extreme directory nesting in swapper

### CODE REVIEW

**Review by 0xApotheosis — COMMENTED** [review]
Conceptually excellent, though I haven't tested against a `web` branch yet.

Nice singular naming for the types 👌 

Also this will probably be a breaking change, so we need to be sure to add a breaking change commit before merging.

**Comment by kaladinlight** [comment]
> Conceptually excellent, though I haven't tested against a `web` branch yet.
> 
> Nice singular naming for the types ok_hand
> 
> Also this will probably be a breaking change, so we need to be sure to add a breaking change commit before merging.

good call on breaking change, will update and open for official review

File: packages/chain-adapters/src/utxo/UtxoBaseAdapter.ts [hunk 1]
```diff
@@ -51,6 +52,12 @@ export const utxoChainIds = [
 
 export type UtxoChainId = typeof utxoChainIds[number]
 
+export type UtxoChainAdapter =
+  | bitcoin.ChainAdapter
+  | bitcoincash.ChainAdapter
+  | dogecoin.ChainAdapter
+  | litecoin.ChainAdapter
+
```
Comment (gomesalexandre) [inline packages/chain-adapters/src/utxo/UtxoBaseAdapter.ts:60]: 💜

File: packages/swapper/src/swappers/thorchain/getThorTradeQuote/getTradeQuote.ts [hunk 2]
```diff
@@ -187,9 +192,9 @@ export const getThorTradeQuote: GetThorTradeQuote = async ({ deps, input }) => {
           }
         })()
       case CHAIN_NAMESPACE.CosmosSdk:
-        return (async (): Promise<TradeQuote<KnownChainIds.CosmosMainnet>> => {
+        return (async (): Promise<TradeQuote<ThorCosmosSdkSupportedChainId>> => {
           const feeData = await (
-            sellAdapter as ChainAdapter<KnownChainIds.CosmosMainnet>
+            sellAdapter as unknown as CosmosSdkBaseAdapter<ThorCosmosSdkSupportedChainId>
```
Comment (gomesalexandre) [inline packages/swapper/src/swappers/thorchain/getThorTradeQuote/getTradeQuote.ts:197]: 🐐

**Review by gomesalexandre — APPROVED** [review]
Untested but conceptual stamp, in on the accuracy brought by the new naming and types: this finally brings the *right* swapper types and terminology, not a subset/superset that somehow works compiles, does what it should (or does it?) at runtime but isn't accurate and is a recipe for bugs to pop up.

Happy to go with this one and bump web in a matching PR

**Comment by kaladinlight** [comment]
https://github.com/shapeshift/web/pull/3781 web pr ready on second stamp

**Comment by kaladinlight** [comment]
lol... just had to close comments

**Pull request merged**